### DISCUSSION TITLE
Error while modifying piggy banks

### DISCUSSION THREAD

**Discussion by honduken** [firefly-iii/firefly-iii#8398] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I was modifying a piggy bank and received this error message. The error message happened upon me selecting "Save" 

### Debug information

Debug information generated at 2024-01-15 09:49:46 for Firefly III version **v6.1.6**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>v6.1.6 / v2.0.12 / 22 (exp. 22)</td></tr><tr><td>PHP version</td><td>8.3.1 / apache2handler / Linux</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>2097152 (2 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, </td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>America/Indiana/Indianapolis + America/New_York</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>smtp</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has piggy banks">:pig:</span> <span title="Has rules">:wrench:</span> <span title="Has subscriptions">:email:</span></td></tr><tr><td>Session start</td><td>2024-01-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-01-31 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36</td></tr></tbody></table>

### Expected behaviour

I expected it to save my piggy bank information and go back to the piggy bank main page 

### Steps to reproduce

I had modified a piggy bank end date value, then thought I should put it further into the future, so went back into it fairly quickly after updating, and then attempted to update it again, and then received this error. I think the issue is related to the speed at which I was performing the action. 

### Additional info

I am now no longer able to utilize the site for anything as every page results in this error message. 

I am unable to reproduce utilizing the demo site. 

I would say it's a me issue however I have 3 other web servers going on the same server that don't have issues with and utilize php and mysql. Also, it was working prior to this happening. If there was an issue with writing to the SQL server or something I would expect just an error stating the record could not be saved and not have the site break completely.

**Comment by honduken** [comment]
Too add more context to this as well, I have automated emails set up for whenever an error occurs, and this is what the email is stating: 

 Hi there,

Firefly III v6.1.6 ran into an error: file_put_contents(/var/www/firefly/budget/storage/framework/cache/data/22/64/2264380afcb078255beff161c0ec65a89b1d5bf8): Failed to open stream: No such file or directory.

The error was of type "ErrorException".

The error occurred on/at: Mon, 15 Jan 2024 09:49:07 -0500.

This error occurred in file "/var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Filesystem/Filesystem.php" on line 205 with code 0. 

 The full stacktrace is below:

#0 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(255): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
#1 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}()
#2 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Filesystem/Filesystem.php(205): file_put_contents()
#3 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Cache/FileStore.php(83): Illuminate\Filesystem\Filesystem->put()
#4 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Cache/FileStore.php(207): Illuminate\Cache\FileStore->put()
#5 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Cache/Repository.php(367): Illuminate\Cache\FileStore->forever()
#6 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(429): Illuminate\Cache\Repository->forever()
#7 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(355): Illuminate\Cache\CacheManager->__call()
#8 /var/www/firefly/budget/app/Support/CacheProperties.php(81): Illuminate\Support\Facades\Facade::__callStatic()
#9 /var/www/firefly/budget/app/Support/Amount.php(117): FireflyIII\Support\CacheProperties->store()
#10 /var/www/firefly/budget/app/Support/Amount.php(100): FireflyIII\Support\Amount->getDefaultCurrencyByUserGroup()
#11 /var/www/firefly/budget/app/Http/Middleware/Range.php(116): FireflyIII\Support\Amount->getDefaultCurrency()
#12 /var/www/firefly/budget/app/Http/Middleware/Range.php(50): FireflyIII\Http\Middleware\Range->configureView()
#13 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\Http\Middleware\Range->handle()
#14 /var/www/firefly/budget/vendor/jc5/google2fa-laravel/src/Middleware.php(29): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#15 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): PragmaRX\Google2FALaravel\Middleware->handle()
#16 /var/www/firefly/budget/app/Http/Middleware/Authenticate.php(66): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#17 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\Http\Middleware\Authenticate->handle()
#18 /var/www/firefly/budget/vendor/laravel/passport/src/Http/Middleware/CreateFreshApiToken.php(63): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#19 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Laravel\Passport\Http\Middleware\CreateFreshApiToken->handle()
#20 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Session/Middleware/AuthenticateSession.php(60): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#21 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Session\Middleware\AuthenticateSession->handle()
#22 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#23 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle()
#24 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#25 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\View\Middleware\ShareErrorsFromSession->handle()
#26 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#27 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\Session\Middleware\StartSession->handleStatefulRequest()
#28 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Session\Middleware\StartSession->handle()
#29 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#30 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle()
#31 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#32 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Cookie\Middleware\EncryptCookies->handle()
#33 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#34 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Routing/Router.php(805): Illuminate\Pipeline\Pipeline->then()
#35 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Routing/Router.php(784): Illuminate\Routing\Router->runRouteWithinStack()
#36 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Routing/Router.php(748): Illuminate\Routing\Router->runRoute()
#37 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Routing/Router.php(737): Illuminate\Routing\Router->dispatchToRoute()
#38 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(200): Illuminate\Routing\Router->dispatch()
#39 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}()
#40 /var/www/firefly/budget/app/Http/Middleware/InstallationId.php(47): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#41 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\Http\Middleware\InstallationId->handle()
#42 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#43 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Http\Middleware\TrustProxies->handle()
#44 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#45 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
#46 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull->handle()
#47 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#48 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
#49 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\TrimStrings->handle()
#50 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#51 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle()
#52 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(99): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#53 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance->handle()
#54 /var/www/firefly/budget/app/Http/Middleware/SecureHeaders.php(48): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#55 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\Http\Middleware\SecureHeaders->handle()
#56 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#57 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(175): Illuminate\Pipeline\Pipeline->then()
#58 /var/www/firefly/budget/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(144): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter()
#59 /var/www/firefly/budget/public/index.php(73): Illuminate\Foundation\Http\Kernel->handle()
#60 {main}

**Comment by honduken** [comment]
I was able to browse to /flush to fix the issue as mentioned on #3456 however this is the second time this has happened so I will keep this open as it does appear to be an issue for me.

**Comment by JC5** [comment]
- https://docs.firefly-iii.org/references/faq/docker/#i-get-permission-denied-errors-on-the-cache-folder
- https://docs.firefly-iii.org/references/faq/install/#unable-to-write-to-cache-directory

**Comment by honduken** [comment]
Yeah, I have done that, it doesn't fix my issue. The only way to fix the issue is to do a /flush command, which is stated already.
